src/node_zlib.cc

```diff
@@ -904,7 +909,20 @@ class ZstdStream final : public CompressionStream<CompressionContext> {
     }
 
     AllocScope alloc_scope(wrap);
-    CompressionError err = wrap->context()->Init(pledged_src_size);
+    std::string_view dictionary;
+    if (args.Length() == 5 && !args[4]->IsUndefined()) {
+      if (!args[4]->IsArrayBufferView()) {
+        THROW_ERR_INVALID_ARG_TYPE(wrap->env(),
+          "dictionary must be an ArrayBufferView if provided");
+        return;
+      }
+
+      ArrayBufferViewContents<char> contents(args[4].As<v8::ArrayBufferView>());
```

Comment:
```suggestion
    ArrayBufferViewContents<char> contents;
    if (args.Length() == 5 && !args[4]->IsUndefined()) {
      if (!args[4]->IsArrayBufferView()) {
        THROW_ERR_INVALID_ARG_TYPE(wrap->env(),
          "dictionary must be an ArrayBufferView if provided");
        return;
      }

      contents.ReadValue(args[4]);
```

`contents` may allocate temporary storage for the contents of the buffer, so it needs to live at least as long as `dictionary`